### PR TITLE
Reduce UriHelper to static methods.

### PR DESCRIPTION
**Pull request by Tratcher**
#275

### CODE REVIEW

**Comment by Tratcher** [comment]
@muratg @davidfowl

File: src/Microsoft.AspNet.Http.Extensions/UriHelper.cs [hunk 1]
```diff
@@ -8,107 +8,46 @@ namespace Microsoft.AspNet.Http.Extensions
...
-            return Scheme + "://" + Host + path + Query + Fragment;
-        }
-
-        public override string ToString()
-        {
-            return IsFullUri ? GetFullUri() : GetPartialUri();
-        }
-
-        public static string Create(PathString pathBase,
+        public static string Encode(PathString pathBase,
```
Comment (davidfowl) [inline src/Microsoft.AspNet.Http.Extensions/UriHelper.cs]: Doc comment? What does this do exactly

**Comment by Tratcher** [comment]
Updated.

File: src/Microsoft.AspNet.Http.Extensions/UriHelper.cs [hunk 2]
```diff
@@ -8,107 +8,68 @@ namespace Microsoft.AspNet.Http.Extensions
...
-
-        public override string ToString()
-        {
-            return IsFullUri ? GetFullUri() : GetPartialUri();
-        }
-
-        public static string Create(PathString pathBase,
+        /// <summary>
+        /// Combines the given uri components into a string that is properly encoded for use in HTTP headers.
+        /// </summary>
```
Comment (muratg) [inline src/Microsoft.AspNet.Http.Extensions/UriHelper.cs]: Doc nit: lowercase "uri" should probably be URI (or Uri)

File: src/Microsoft.AspNet.Http.Extensions/ResponseHeaders.cs [hunk 3]
```diff
@@ -126,20 +126,20 @@ public EntityTagHeaderValue ETag
...
                 {
-                    return new UriHelper(uri);
+                    return uri;
                 }
                 return null;
             }
             set
             {
-                Headers.Set(HeaderNames.Location, value);
+                Headers.Set(HeaderNames.Location, value == null ? null : UriHelper.Encode(value));
```
Comment (muratg) [inline src/Microsoft.AspNet.Http.Extensions/ResponseHeaders.cs]: Consider using null coalescing (?? operator)

File: src/Microsoft.AspNet.Http.Extensions/UriHelper.cs [hunk 4]
```diff
@@ -8,107 +8,68 @@ namespace Microsoft.AspNet.Http.Extensions
...
-                Query = query,
-                Fragment = fragment
-            }.GetPartialUri();
+            string combinePath = (pathBase.HasValue || path.HasValue) ? (pathBase + path).ToString() : "/";
+            return combinePath + query + fragment;
         }
 
-        public static string Create(string scheme,
+        /// <summary>
+        /// Combines the given uri components into a string that is properly encoded for use in HTTP headers.
```
Comment (muratg) [inline src/Microsoft.AspNet.Http.Extensions/UriHelper.cs]: Ditto

File: src/Microsoft.AspNet.Http.Extensions/ResponseHeaders.cs (hunk 3 above)
Comment (Tratcher) [inline src/Microsoft.AspNet.Http.Extensions/ResponseHeaders.cs]: That only works for the reverse case `value == null ? DoSomethingElse() : null;` We only want to call Encode if it is not null.

Comment (muratg) [inline src/Microsoft.AspNet.Http.Extensions/ResponseHeaders.cs]: Ah, right.

**Comment by muratg** [comment]
:shipit:

**Pull request merged**